latex.bzl

```diff
@@ -31,19 +44,29 @@ def _latex_pdf_impl(ctx):
             ],
         ),
         outputs = [ctx.outputs.out],
-        tools = [ctx.executable.tool],
+        tools = [ctx.executable._tool],
     )
+    latex_info = LatexOutputInfo(file = ctx.outputs.out, format=ext)
+    return [latex_info]
 
-_latex_pdf = rule(
+_latex = rule(
     attrs = {
-        "main": attr.label(allow_files = True),
+        "main": attr.label(
+            allow_single_file = [".tex"],
+            mandatory = True,
+         ),
         "srcs": attr.label_list(allow_files = True),
         "cmd_flags": attr.string_list(
             allow_empty = True,
             default = [],
         ),
-        "tool": attr.label(
-            default = Label("//:run_lualatex"),
+        "format": attr.string(
+            doc = "Output file format",
+            default = "pdf",
+            values = ["dvi","pdf"],
```

Comment:
Very much nitpicking there, but if you have [`buildifier`](https://github.com/bazelbuild/buildtools/tree/master/buildifier) installed, consider running it - I believe there should be a space between `"dvi",` and `"pdf"`.

Reply:
I'm afraid not...
I do most of the edit off of my mobile currently (hence the many commits :D ).
But I can fix this part :)

but as a compromise,
we could add buildifier to the pipeline in another issue:
https://github.com/bazelbuild/buildtools/blob/master/buildifier/README.md#setup-and-usage-via-bazel-not-supported-on-windows
E.g.
bazel run //:buildifier -- --lint=warn

Reply:
Sounds good - thanks for opening the issue!